From 7e8ec726e892b16e394b711e22a29424a3c04315 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Fri, 4 Oct 2024 08:35:54 +0800
Subject: [PATCH 123/206] mtk: hostapd: add return value for
 ieee802_11_set_beacon_per_bss_only

Since __ieee802_11_set_beacon() can fail, the
ieee802_11_set_beacon_per_bss_only should forward the return value to
its caller.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/beacon.c | 4 ++--
 src/ap/beacon.h | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ap/beacon.c b/src/ap/beacon.c
index 8ad6ed56c..603ed02da 100644
--- a/src/ap/beacon.c
+++ b/src/ap/beacon.c
@@ -2882,9 +2882,9 @@ fail:
 }
 
 
-void ieee802_11_set_beacon_per_bss_only(struct hostapd_data *hapd)
+int ieee802_11_set_beacon_per_bss_only(struct hostapd_data *hapd)
 {
-	__ieee802_11_set_beacon(hapd);
+	return __ieee802_11_set_beacon(hapd);
 }
 
 
diff --git a/src/ap/beacon.h b/src/ap/beacon.h
index a3c8d53a3..19033b5a2 100644
--- a/src/ap/beacon.h
+++ b/src/ap/beacon.h
@@ -50,7 +50,7 @@ enum {
 void handle_probe_req(struct hostapd_data *hapd,
 		      const struct ieee80211_mgmt *mgmt, size_t len,
 		      int ssi_signal);
-void ieee802_11_set_beacon_per_bss_only(struct hostapd_data *hapd);
+int ieee802_11_set_beacon_per_bss_only(struct hostapd_data *hapd);
 int ieee802_11_set_beacon(struct hostapd_data *hapd);
 int ieee802_11_set_beacons(struct hostapd_iface *iface);
 int ieee802_11_set_bss_critical_update(struct hostapd_data *hapd,
-- 
2.45.2

